File: 07-02-2025/Cpp/ReverseInteger.cpp
//https://leetcode.com/problems/reverse-integer/description//

class Solution {
public:
    int reverse(int x) {
        int result = 0;
        if (x == INT_MIN) {
            return 0;
        }
        int sign = (x < 0) ? -1 : 1;  
         x = abs(x);
        
        while (x != 0) {
            int digit = x % 10;
            x /= 10;
            
            if (result > INT_MAX / 10 || (result == INT_MAX / 10 && digit > 7)) {
                return 0;  
            }
            
            result = result * 10 + digit;
        }
        
        return sign * result;  
    }
};

/*  TEST CASES

Input: x = 123
Output: 321

Input: x = -123
Output: -321


Input: x = 120
Output: 21
*/

File: 07-02-2025/Java/Reverse Integer.java
//https://leetcode.com/problems/reverse-integer/description/

class Solution {
    public int reverse(int x) {
        long ans = 0;

        while(x != 0) {
            int digit = x % 10;

            x /= 10;

            ans = (ans * 10) + digit;
        }

        if(ans > Integer.MAX_VALUE || ans < Integer.MIN_VALUE) {
            return 0;
        }

        return (int)ans;
   }
}



// Example 1:

// Input: x = 123
// Output: 321
